RFC/src/RFC-0151_TransactionProtocol.md

```diff
@@ -0,0 +1,255 @@
+and _m_ is the additional metadata for the transaction. \\(\Sigma P_i\\) is the value of the (pre-offset) excess that is
+stored in the transaction kernel.
+
+Notice that every signing party needs to know the sum of all the nonces and public spending keys. This suggests that
+every party knows how many parties are involved in the transaction, which is not an ideal privacy scenario. It would be
+preferable if a secure scheme could be found where each recipient interacts only with the sender and does not need to
+calculate these sums themselves.
+
+Unfortunately, as we discuss below, it seems that it's not possible using any known scheme that satisfies both this
+privacy goal while achieving the desired privacy level.
+
+
+### The issue with multiple senders
+
+To increase privacy, the public excess values are [offset] by a constant random value. The choice of this value, as well
+as fee selection can only be set once per transaction. The privilege of selecting these values is generally bestowed on
+the sender, since she pays the fee. Allowing multiple sending parties (or equivalently, allowing recipients to provide
+inputs) would require a negotiation round to set the fee and offset before the transaction could be constructed. This is
+a complication we don't want to deal with, and so all schemes presented here allow exactly one sender.
+
+### Two-party transactions
+
+Two party transactions are fairly straightforward and are described in detail on TLU. (See [Mimblewimble transaction]).
+
+It is proposed that Tari implement this single-round 2-party transaction scheme as a special case to support both online
+2-party transactions as well as "offline" transactions such as via e-mail, text-message, carrier pigeon etc.
+
+
+###  Multiple recipient transaction scheme
+
+<div class="mermaid">
+sequenceDiagram
+    participant Sender
+    participant Receivers
+# Sender Initializes the transaction
+    activate Sender
+        Sender-->>Sender: initialize
+    deactivate Sender
+# Sender transmits initial data to all receivers
+    activate Sender
+        Sender-->>+Receivers: [tx_id, amt_i,H(Rs||Xs),m]
+        note left of Sender: CollectingCommitments
+        note right of Receivers: SendingCommitment
+        Receivers-->>-Sender: [tx_id, H(Ri||Pi)]
+     deactivate Sender
+# Receiving Outputs
+     activate Sender
+        Sender-->>+Receivers: [tx_id, [Hashed commitments]]
+        note left of Sender: CollectingPubKeys
+        note right of Receivers: SendingOutput
+        Receivers-->>-Sender: [tx_id, Pi, Ri, C_i, RP_i]
+    deactivate Sender
+    alt inflation_error()?
+        Sender--XReceivers: [tx-id, failed]
+    end
+# Signature collection
+    activate Sender
+        Sender-->>+Receivers: [tx_id, [Rs, Ri] [Xs, Pi]]
+        note left of Sender: CollectingSignatures
+        note right of Receivers: Signing
+        Receivers-->>Receivers: validate nonce and pubkey sum
```

Comment:
Do you mean the Sender's nonce, Rs, and excess, Xs, (is this what you mean by public key sum) are validated against the hash sent in the first step, H(Rs||Xs)?

Reply:
Or do you mean checking all the respective nonces and keys against the the individual H(Ri||Pi) terms? It's the `pubkey sum` that is confusing me.